lib/cc/analyzer/container.rb

```diff
@@ -128,10 +123,18 @@ def read_stderr(err)
 
       def timeout_thread
         Thread.new do
-          sleep timeout
+          # Doing one long `sleep timeout` seems to fail sometimes, so
+          # we do a series of short timeouts before exiting
+          start_time = Time.now
+          loop do
+            sleep 1
+            duration = Time.now - start_time
+            break if duration >= timeout
+          end
+
           @timed_out = true
           reap_running_container
```

Comment:
Interesting idea - I would like to ship this and revisit that if the problem isn't solved, but keeping this as small as possible seems like a good idea